pkg/logs/decoder/line_handler.go

```diff
@@ -228,6 +230,10 @@ func (h *MultiLineHandler) sendContent() {
 			log.Debug(err)
 		}
 		if output != nil && len(output.Content) > 0 {
+			// The output.Timestamp filled by the Parse function is the ts of the first
+			// log line, in order to be useful to setLastSince function, we need to replace
+			// it with the ts of the last log line
+			output.Timestamp = h.lastSeenTS
```

Comment:
Good catch, the only concern I have here is that we are adding a docker logic inside the line_handler which is supposed to be agnostic from inputs.
From a feature point of view, it is also counter intuitive to set the timestamp to the last line seen, we would expect the first line ; this change breaks this logic.

Reply:
Can't come up with something better, but I'd like to comment to be clear about what's the use of `output.Timestamp`, it is not meant to stamp the log but to recover well when tailing back the container.

Reply:
agreed. Added to the comments